app/services/user/service.go

```diff
@@ -170,6 +185,49 @@ func (u *userService) CreateEvent(ctx context.Context, orgUser *userRepo.GetUser
 	return nil
 }
 
+func (u *userService) UpdateEvent(ctx context.Context, orgUser *userRepo.GetUserWithOrgRow, eventParams *dto.EventUpdate, eventID uuid.UUID) error {
+	policyNumber := bo.PolicyNumber(orgUser.PoliciesNum)
+	if !policyNumber.Can(bo.MODIFY_EVENTS) {
+		return errors.New("user cannot modify events")
+	}
+
+	imageUrl, err := u.storage.Put(ctx, fmt.Sprintf("users/%s.webp", eventParams.EventID), &objectstore.Object{
+		Data: eventParams.EventImage,
+	})
+	if err != nil {
+		log.Errorf("%v", err)
+		return err
+	}
+
+	if err := u.userRepo.UpdateEvent(ctx, &userRepo.UpdateEventParams{
+		ID:               eventID,
+		EventName:        eventParams.EventName,
+		EventDate:        eventParams.EventDate,
+		EventLocation:    eventParams.EventLocation,
+		EventDescription: eventParams.EventDescription,
+		EventImageUrl:    imageUrl,
+	}); err != nil {
+		log.Errorf("%v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (u *userService) DeleteEvent(ctx context.Context, orgUser *userRepo.GetUserWithOrgRow, eventID uuid.UUID) error {
+	policyNumber := bo.PolicyNumber(orgUser.PoliciesNum)
+	if !policyNumber.Can(bo.CREATE_EVENTS) {
+		return errors.New("user cannot create or delete events")
+	}
+
+	if err := u.userRepo.DeleteEvent(ctx, eventID); err != nil {
+		log.Errorf("%v", err)
+		return err
+	}
+
+	return nil
+}
```

Comment:
This goes for every method you add, but make sure that you unit test it beforehand. Thx.